hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -303,7 +296,37 @@ public StorageStatistics provide() {
     } catch (AmazonClientException e) {
       throw translateException("initializing ", new Path(name), e);
     }
+  }
+
+  private int getMaxThreads(Configuration conf, String maxThreadsKey, int defaultMaxThreads) {
+    int maxThreads = conf.getInt(maxThreadsKey, defaultMaxThreads);
+    if (maxThreads < 2) {
+      LOG.warn(maxThreadsKey + " must be at least 2: forcing to 2.");
+      maxThreads = 2;
+    }
+    return maxThreads;
+  }
+
+  private LazyTransferManager createLazyUploadTransferManager(Configuration conf) {
```

Comment:
this logic could be moved into LazyUploadTransferManager itself, maybe just as a static method

Reply:
Done